src/integrationTest/java/io/aiven/kafka/connect/jdbc/oracle/VerifyInsertIT.java

```diff
@@ -0,0 +1,189 @@
+            + "    \"value\" VARCHAR2(255) NOT NULL,\n"
+            + "PRIMARY KEY(\"id\")"
+            + ")", TEST_TOPIC_NAME);
+    private static final String CREATE_TABLE = String.format("CREATE TABLE \"%s\" (\n"
+            + "    \"id\" NUMBER NOT NULL,\n"
+            + "    \"name\" VARCHAR2(255) NOT NULL,\n"
+            + "    \"value\" VARCHAR2(255) NOT NULL\n"
+            + ")", TEST_TOPIC_NAME);
+
+
+    private Map<String, String> basicSinkConnectorConfig() {
+        final Map<String, String> config = basicConnectorConfig();
+        config.put("name", CONNECTOR_NAME);
+        config.put("connector.class", JdbcSinkConnector.class.getName());
+        config.put("topics", TEST_TOPIC_NAME);
+        return config;
+    }
+
+    private Map<String, String> sinkConnectorConfigWithPKModeRecordKey() {
+        final Map<String, String> config = basicSinkConnectorConfig();
+        config.put("pk.mode", "record_key");
+        config.put("pk.fields", "id"); // assigned name for the primitive key
+        return config;
+    }
+
+    private ProducerRecord<String, GenericRecord> createRecord(
+            final int id, final int partition, final String name, final String value) {
+        final GenericRecord record = new GenericData.Record(VALUE_RECORD_SCHEMA);
+        record.put("id", id);
+        record.put("name", name);
+        record.put("value", value);
+        return new ProducerRecord<>(TEST_TOPIC_NAME, partition, String.valueOf(id), record);
+    }
+
+    private void sendTestData(final int numberOfRecords) throws InterruptedException, ExecutionException {
+        final List<Future<RecordMetadata>> sendFutures = new ArrayList<>();
+        for (int i = 0; i < numberOfRecords; i++) {
+            for (int partition = 0; partition < TEST_TOPIC_PARTITIONS; partition++) {
+                final String recordName = "user-" + i;
+                final String recordValue = "value-" + i;
+                final ProducerRecord<String, GenericRecord> msg = createRecord(i, partition, recordName, recordValue);
+                sendFutures.add(producer.send(msg));
+            }
+        }
+        producer.flush();
+        for (final Future<RecordMetadata> sendFuture : sendFutures) {
+            sendFuture.get();
+        }
+    }
+
+    @AfterEach
+    public void afterEach() throws SQLException {
+        executeSqlStatement(DROP_TABLE);
+    }
+
+    @Test
+    public void testSinkConnector() throws Exception {
+        createTopic(TEST_TOPIC_NAME, 1); // Create Kafka topic matching the table name
+        executeSqlStatement(CREATE_TABLE);
+        consumer.assign(Collections.singleton(new TopicPartition(TEST_TOPIC_NAME, 0)));
+        // Start the sink connector
+        connectRunner.createConnector(basicSinkConnectorConfig());
+
+        // Send test data to Kafka topic
+        sendTestData(1);
+
+        await().atMost(Duration.ofSeconds(20)).pollInterval(Duration.ofSeconds(19))
```

Comment:
Just curious, why specify a poll interval of 19 seconds? Wouldn't we want to poll more frequently like once a second in order for the assertion (and, as a result, the tests) to pass sooner?